ocaml/xapi/cert_distrib.ml

```diff
@@ -403,7 +420,6 @@ let exchange_certificates_in_pool ~__context =
     | Some seed ->
         fun ops ->
           Random.init seed ;
-          let rand_i = Random.int (List.length ops) in
```

Comment:
Isn't it enough to use `List.length ops - 1` here?

Reply:
I don't like the existing function to take a position where it inserts - this should be the responsibility of the function. I think the existing semantics are not very clear and that is why I rewrote it.

Reply:
I actually thought the old function was easier to understand than the new one.

Reply:
Actually... The bound of `Random.int` is excluded, so we should already not take the last element. So why is this wrong or is there another problem?